### PR TITLE
Remove device, firmware, location and connectivity objects from the client daemon

### PR DESCRIPTION
**Pull request by seank-img**
Object definitions for the firmware, location and connectivity objects should now be loaded when the daemon starts using the ```--objDefs``` or ```-o``` option.

### CODE REVIEW

File: api/tests/object-defs-gtest.xml [hunk 1]
```diff
@@ -0,0 +1,530 @@
+<ObjectDefinitions>
+	<Items>
```
Comment (DavidAntliff) [inline api/tests/object-defs-gtest.xml]: maybe replace tabs with spaces?

File: api/tests/support/process.h [hunk 2]
```diff
@@ -62,7 +62,7 @@ bool IsUDPPortInUse(int port);
 int WaitForIpc(int ipcPort, int timeout /*seconds*/, const char * request, size_t requestLen);
 
 // Start an Awa Client process on the specified CoAP and IPC port. Redirect output to logFile. Return process ID, or 0 if failed.
-pid_t StartAwaClient(const char * clientDaemonPath, int coapPort, int ipcPort, const char * logFile, const char * clientID, const char * bootstrapConfig, const char * bootstrapURI, const std::vector<std::string> & additionalOptions);
+pid_t StartAwaClient(const char * clientDaemonPath, int iCoapPort, int iIpcPort, const char * logFile, const char * endpointName, const char * bootstrapConfig, const char * bootstrapURI, const char * objectDefinitionsFile, const std::vector<std::string> & additionalOptions);
```
Comment (DavidAntliff) [inline api/tests/support/process.h]: Prior to your change, additionalOptions was used to pass object definition files to the daemon. It makes sense to explicitly pass an object definitions file here, but keep in mind that there may be multiple files (daemon already supports multiple files). Consider changing `objectDefinitionsFile` to a std::vector<std::string> of object definition filenames.

File: tools/tests/object-defs-tools-test.xml [hunk 3]
```diff
@@ -0,0 +1,530 @@
+<ObjectDefinitions>
+	<Items>
```
Comment (DavidAntliff) [inline tools/tests/object-defs-tools-test.xml]: tabs

**Comment by DavidAntliff** [comment]
@seank-img, looks like a significant chunk of work - great that you did this, thank you.

**Comment by DavidAntliff** [comment]
@seank-img do we need some basic object definition files for installation into /etc/awa (or /usr/share/awa)? Is this planned for a future PR? Would be good to have all the standard objects and maybe even IPSO objects available for users to use. We'd need to think how we integrate this with systemd - may need to use a /etc/awa/objects.d/<files> approach.

**Comment by seank-img** [comment]
@DavidAntliff  yes, I've added those objects to the config dir - for now it's probably alright as it is. Future PRs could add the IPSO objects, and appropriate system.d support. Is there anything documentation wise that I need to do?

**Comment by DavidAntliff** [comment]
@seank-img do the client/server daemons pick up the standard object definition files automatically? If not, you may need to document that change. Also, after a 'make install', do the object definition files get copied somewhere like /etc or /usr/share? If not, that probably needs to be done, and documented, otherwise the 'installed' version won't be usable. If there's no current automatic loading, I suggest they could be auto-loaded from /etc/awa/objects.d/*.xml, or stored in /usr/share/awa/objects/*.xml and the init.d script updated to load them explicitly.

**Comment by seank-img** [comment]
retest AWS CI (erbium)

**Comment by delme-imgtec** [comment]
retest AWS CI (erbium)

**Pull request merged**